crates/core/src/db/relational_db.rs

```diff
@@ -2591,7 +2591,40 @@ mod tests {
     }
 
     #[test]
-    fn test_view_tables_are_ephemeral() -> ResultTest<()> {
+    fn test_view_tables_are_ephemeral_in_commitlog() -> ResultTest<()> {
+        let stdb = TestDB::durable_without_snapshot_repo()?;
+
+        let (view_id, table_id, _, _) = setup_view(&stdb)?;
+
+        // Write some rows (reusing the same helper)
+        insert_view_row(&stdb, view_id, table_id, Identity::ONE, 10)?;
+        insert_view_row(&stdb, view_id, table_id, Identity::ZERO, 20)?;
+
+        assert!(
+            !project_views(&stdb, table_id, Identity::ZERO).is_empty(),
+            "View table should NOT be empty after insert"
+        );
+
+        // Reopen the database — view tables must not persist
+        let stdb = stdb.reopen()?;
```

Comment:
I believe flakiness was coming from this [patch]( https://github.com/clockworklabs/SpacetimeDB/blob/3e7ab2ccac96fe4257d6bf222440cc3d190a87b0/crates/core/src/db/relational_db.rs#L1784-L1792) which gets called from `reopen`, code snippet is trying to take a snapshot and then calling `open_db` instantly, which might have or have not snapshot created by the time.

I splitted the tests in two - `test_view_tables_are_ephemeral_in_commitlog:` which explicitly does not create snapshot repo.


`test_view_tables_are_ephemeral_with_snapshot`: which calls method `stdb.take_snapshot` before reopening db.